src/Parser/Warlock/Affliction/Modules/Features/Abilities.js

```diff
@@ -8,135 +8,119 @@ class Abilities extends CoreAbilities {
-        cooldown: 40,
-        enabled: combatant.hasTalent(SPELLS.PHANTOM_SINGULARITY_TALENT.id),
-        isOnGCD: true,
-        castEfficiency: {
-          suggestion: true,
-        },
-      },
       {
         spell: SPELLS.AGONY,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        category: Abilities.SPELL_CATEGORIES.DOTS,
         isOnGCD: true,
       },
       {
         spell: SPELLS.CORRUPTION_CAST,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        category: Abilities.SPELL_CATEGORIES.DOTS,
         isOnGCD: true,
       },
       {
         spell: SPELLS.SIPHON_LIFE_TALENT,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        category: Abilities.SPELL_CATEGORIES.DOTS,
         enabled: combatant.hasTalent(SPELLS.SIPHON_LIFE_TALENT.id),
         isOnGCD: true,
       },
       {
         spell: SPELLS.UNSTABLE_AFFLICTION_CAST,
-        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        category: Abilities.SPELL_CATEGORIES.DOTS,
         isOnGCD: true,
       },
       {
-        spell: SPELLS.DRAIN_SOUL,
+        spell: SPELLS.DEATHBOLT_TALENT,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
         isOnGCD: true,
+        enabled: combatant.hasTalent(SPELLS.DEATHBOLT_TALENT.id),
+        cooldown: 30,
+        castEfficiency: {
+          suggestion: true,
+          recommendedEfficiency: 0.9,
+        },
       },
       {
-        spell: SPELLS.LIFE_TAP,
+        spell: SPELLS.HAUNT_TALENT,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        cooldown: 15,
+        enabled: combatant.hasTalent(SPELLS.HAUNT_TALENT.id),
         isOnGCD: true,
+        castEfficiency: {
+          suggestion: true,
+          recommendedEfficiency: 0.95,
+          // TODO: possibly implement Haunt resets via SpellUsable?
+          extraSuggestion: 'This estimate may not be correct sometimes because of Haunt\'s resets. The real amount of possible Haunts will be higher if there were adds on this fight.',
+        },
       },
       {
-        spell: SPELLS.REAP_SOULS,
-        category: Abilities.SPELL_CATEGORIES.UTILITY,
-        cooldown: 5,
+        spell: SPELLS.SHADOW_BOLT_AFFLI,
+        category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        isOnGCD: true,
+        enabled: !combatant.hasTalent(SPELLS.DRAIN_SOUL_TALENT.id),
       },
       {
-        spell: SPELLS.SEED_OF_CORRUPTION_DEBUFF,
+        spell: SPELLS.DRAIN_SOUL_TALENT,
         category: Abilities.SPELL_CATEGORIES.ROTATIONAL,
+        enabled: combatant.hasTalent(SPELLS.DRAIN_SOUL_TALENT.id),
         isOnGCD: true,
       },
-
-      // Cooldowns
       {
-        spell: SPELLS.SOUL_HARVEST_TALENT,
-        category: Abilities.SPELL_CATEGORIES.COOLDOWNS,
-        cooldown: 120,
-        enabled: combatant.hasTalent(SPELLS.SOUL_HARVEST_TALENT.id),
+        spell: SPELLS.PHANTOM_SINGULARITY_TALENT,
+        category: Abilities.SPELL_CATEGORIES.ROTATIONAL_AOE,
```

Comment:
If you're supposed to be casting this on cooldown on single target I'd suggest changing it to `ROTATIONAL` (or if you feel it is more appropriate, `COOLDOWN`) instead. If not, I'd suggest disabling castEfficiency suggestion.

This goes for all the other AOE rotational spells aswell.

Reply:
It depends. On Legion, many of us were running PS even on single target, casting it on cooldown, but it *is* better suited for recurring add spawns, picture something like Portal Keeper. So it isn't generally a cooldown, but I wouldn't necessarily call it inherently rotational. Disabling the suggestion might be a good thing though.